WalletWasabi.Fluent/ViewModels/Wallets/Send/PrivacySuggestionsFlyoutViewModel.cs

```diff
@@ -58,8 +58,13 @@ public async Task BuildPrivacySuggestionsAsync(Wallet wallet, TransactionInfo in
 
 		if (hasChange && !isFixedAmount && !info.IsPayJoin)
 		{
-			var suggestions =
-				ChangeAvoidanceSuggestionViewModel.GenerateSuggestionsAsync(info, destination, wallet, linkedCts.Token);
+			int originalInputCount = transaction.SpentCoins.Count();
+
+			// Use at most 5 coins.
+			int maxInputCount = (int)(Math.Max(3, originalInputCount * 1.3));
```

Comment:
Wouldn't `ChangelessTransactionCoinSelector.cs` be a better place for this line?

Reply:
For me, it still feels weird.

Most of the dependencies, that a `StrategyParameters` needs, is built inside `ChangelessTransactionCoinSelector` apart from this one. This one is built inside a `ViewModel`.

We are already sending an `int` through the ViewModels into the `ChangelessTransactionCoinSelector`, so no new dependency would be added. Can't think of any drawbacks on the unit testing side, also.

Anyway, these are only my thoughts. No need to change anything, if you don't think it's worth it.